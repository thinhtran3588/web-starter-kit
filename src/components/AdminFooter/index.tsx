import React from 'react';
import { withTranslation, WithTranslation } from '@app/core';
import { config } from '@app/config';
import { Typography } from '../Typography';
import { Link } from '../Link';
import { useStyles } from './styles';

type Props = WithTranslation;

export const BaseAdminFooter = (props: Props): JSX.Element => {
  const { t } = props;
  const classes = useStyles();

  return (
    <footer className={classes.root}>
      <Typography variant='body1'>
        &copy; <Link href='/'>{config.author}</Link>
        &nbsp;{config.copyRightYear}
      </Typography>
      <Typography variant='caption'>{t('createdWithLove')}</Typography>
    </footer>
  );
};

export const AdminFooter = withTranslation('admin_common')(BaseAdminFooter);
