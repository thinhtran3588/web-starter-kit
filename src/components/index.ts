export * from './AdminLayout';
export * from './AdminFooter';
export * from './Footer';
export * from './Header';
export * from './Icon';
export * from './Layout';
export * from './Sidebar';
export * from './SidebarMenu';
export * from './FormHeader';
export * from './Field';
export * from './FormFilter';
export * from './FormInput';
export * from './FormSearch';
export * from './Table';
export * from './LanguageSelection';
export * from './Link';
export * from './AuthLayout';
export * from './Head';
export * from './Button';
export * from './Grid';
export * from './Typography';
