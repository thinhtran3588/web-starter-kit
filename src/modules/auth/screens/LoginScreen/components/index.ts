export * from './EmailLogin';
export * from './PhoneNoLogin';
