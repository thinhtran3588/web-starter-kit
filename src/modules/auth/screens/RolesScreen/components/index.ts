export * from './Detail';
export * from './PermissionsTable';
