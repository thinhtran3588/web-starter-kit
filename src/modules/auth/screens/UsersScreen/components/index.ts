export * from './Detail';
export * from './PermissionsTable';
export * from './ChangePassword';
