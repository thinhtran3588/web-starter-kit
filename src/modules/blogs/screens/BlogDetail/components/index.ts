export * from './Content';
export * from './SideMenu';
