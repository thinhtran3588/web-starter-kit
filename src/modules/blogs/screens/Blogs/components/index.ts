export * from './Categories';
export * from './BlogsList';
