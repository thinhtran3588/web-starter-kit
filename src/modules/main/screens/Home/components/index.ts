export * from './CoverImage';
export * from './Features';
export * from './Quote';
export * from './Help';
