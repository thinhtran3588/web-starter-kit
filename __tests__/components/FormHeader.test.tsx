import React from 'react';
import { render } from '@testing-library/react';
import { FormHeader } from '@app/components';

describe('components/FormHeader', () => {
  beforeEach(() => {});

  it('renders successfully', async () => {
    const createUser = jest.fn();
    const { baseElement } = render(
      <FormHeader
        breadcrumbLinks={[
          {
            text: 'Auth',
            link: '/admin/users',
          },
          {
            text: 'Users',
          },
        ]}
        commandButtons={[
          {
            text: 'Create',
            onClick: createUser,
            color: 'primary',
          },
        ]}
      />,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
