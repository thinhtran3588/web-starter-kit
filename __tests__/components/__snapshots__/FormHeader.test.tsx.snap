// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FormHeader renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 makeStyles-root-1 MuiPaper-rounded"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiBreadcrumbs-root makeStyles-breadcrumb-2 MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        <ol
          class="MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <next-link
              href="/admin/users"
            >
              <a
                class=""
                href="/admin/users"
              >
                Auth
              </a>
            </next-link>
          </li>
          <li
            aria-hidden="true"
            class="PrivateBreadcrumbSeparator-root-67 MuiBreadcrumbs-separator"
          >
            /
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
            >
              Users
            </p>
          </li>
        </ol>
      </nav>
      <div
        class="makeStyles-commands-3"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-4 MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Create
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;
